assets/js/modules/ads/components/setup/SetupMainPAX.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Ads Main PAX setup component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	createInterpolateElement,
+	Fragment,
+	useCallback,
+} from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import AdsIcon from '../../../../../svg/graphics/ads.svg';
+import SetupForm from './SetupForm';
+import SupportLink from '../../../../components/SupportLink';
+import AdBlockerWarning from '../common/AdBlockerWarning';
+import { SpinnerButton } from 'googlesitekit-components';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import { ADWORDS_SCOPE, MODULES_ADS } from '../../datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import useQueryArg from '../../../../hooks/useQueryArg';
+import { addQueryArgs } from '@wordpress/url';
+import PAXEmbeddedApp from '../PAXEmbeddedApp';
+const { useSelect, useDispatch, useRegistry } = Data;
+
+export default function SetupMainPAX( { finishSetup } ) {
+	const [ showPaxApp, setShowPaxApp ] = useQueryArg( 'pax' );
+
+	const registry = useRegistry();
+
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( CORE_USER ).isAdBlockerActive()
+	);
+	const hasAdwordsScope = useSelect( ( select ) =>
+		select( CORE_USER ).hasScope( ADWORDS_SCOPE )
+	);
+
+	const redirectURL = addQueryArgs( global.location.href, {
+		pax: 1,
+		notification: 'ads', // Arbitrary value to prevent authentication_success from surfacing.
+	} );
+
+	const oAuthURL = useSelect( ( select ) =>
+		select( CORE_USER ).getConnectURL( {
+			additionalScopes: [ ADWORDS_SCOPE ],
+			redirectURL,
+		} )
+	);
+	const isNavigatingToOAuthURL = useSelect( ( select ) => {
+		if ( ! oAuthURL ) {
+			return false;
+		}
+
+		return select( CORE_LOCATION ).isNavigatingTo( oAuthURL );
+	} );
+
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+	const { setPaxConversionID, setExtCustomerID, submitChanges } =
+		useDispatch( MODULES_ADS );
+
+	const onLaunch = useCallback(
+		async ( app ) => {
+			if ( app && typeof app?.getServices === 'function' ) {
```

Comment:
There's no need to have a condition to check for `app` or if `getServices` is present since this function is called only when the app successfully launches. We should assume that it is passed and is an app instance with the expected methods.